### DISCUSSION TITLE
SetOption65 message in console reverted by mistake ?

### DISCUSSION THREAD

**Discussion by rapi3** [arendst/Tasmota#15479] [Q&A] [answered]
Hello,
```
SetOption65	Device recovery using [fast power cycle detection]
(https://tasmota.github.io/docs/Device-Recovery/#fast-power-cycle-device-recovery)
0 = enabled (default)
1 = disabled
```
I am wondering if messages are not reverted by mistake ?
logic is
enable = ON
disabled = OFF
```
21:36:31.515 CMD: so65 0
21:36:31.520 MQT: s/S01/RESULT = {"SetOption65":"OFF"}
21:36:33.821 CMD: so65 1
21:36:33.826 MQT: s/S01/RESULT = {"SetOption65":"ON"}
```

**Comment by barbudor** [comment, marked as answer]
No
Setoption65 0 or Off = enabled
Setoption65 1 or On = disabled
Doc is correct

**Reply by sfromis** [reply]
The value of `SetOption65` is for the question of disabling the standard feature of fast power cycle detection. As this could trigger too much in some situations, an option was introduced to disable this, and generally option bits are off by default. Yeah, that created a double negative, but that's how it is.

**Comment by rapi3** [comment]
**So the programming logic for reporting everything in console is:
0 = OFF
1 = ON**

at least this is a bit confusing only for SO 31, 65, 78
![confused](https://user-images.githubusercontent.com/18531150/165909720-e5500307-324e-4178-a889-f958077d147c.gif)
if you assume 0 = disable and 1 = enable as for everything else...

**Reply by barbudor** [reply]
When a new setoption is introduced, it has to be done in a non breaking way for existing users
So the legacy behavior must be 0 because this is the default initialization value for settings.

**Reply by sfromis** [reply]
> if you assume 0 = disable and 1 = enable as for everything else...

Do not make that bad assumption. What you can set on or off is the feature of disabling the fast power cycle recovery.

More generally, SetOptionNN are not a list of separate commands, it is the command `SetOption` with an index number into an array of bits, to be set on or off (except for the group of options which are number). The command `SetOption` does not "know" what the bit is for, and the code in other parts of Tasmota just checks if a certain bit is set to decide how to proceed. When new options were introduced, they had to have 0/OFF as their backward-compatible initial default value, with 1/ON as the new control available.
